views/search/search.ejs

```diff
@@ -1,61 +1,103 @@
 <form class="container" id="search" method="get">
   <div class="input-group mb-3 search-section row justify-content-center">
     <label for="job_title" class="search-label">Job Title</label>
-
+    <!-- Job title -->
+    <% if(job_title == undefined ){ %>
     <input id="job_title" name="job_title" type="text" class="form-control rounded search-field"
-      placeholder="e.g. Senior Android Developer" aria-label="Username" aria-describedby="basic-addon1">
+      placeholder="e.g. Senior Android Developer" aria-label="Username" aria-describedby="basic-addon1" value="">
+    <% }else { %>
+      <input id="job_title" name="job_title" type="text" class="form-control rounded search-field"
+      placeholder="e.g. Senior Android Developer" aria-label="Username" aria-describedby="basic-addon1" value="<%=job_title %>">
+    <% } %>
 
-    <div class="dropdown search-dropdown">
+    <!-- Number of results -->
+    <div class="dropdown search-dropdown ml-4">
+      <select class="custom-select" id="inputGroupSelect01" name="results">
+        <% 
+        var options = [ "10", "20", "30", "10000" ];
+        var result_labels = [ "10", "20", "30", "All" ];
+        if (size == undefined) { 
+          for ( var i = 0; i < options.length; i++ ){
+          %>
+          <option value="<%=options[i] %>"><%=result_labels[i] %> results</option>
+        <% }} else {
+        var options = [ "10", "20", "30", "10000" ];
+        var result_labels = [ "10", "20", "30", "All" ];
+        for ( var i = 0; i < options.length; i++ )
+        {
+            var selected = (size == options[i]) ? "selected" : "";
```

Comment:
here you declared the variables "options" and "result_labels" twice, and you also loop twice through the same arrays -> could've been shorter by just writing the 2nd loop and put the condition "typeof size !== 'undefined" inside this condition. so basically:
if size is not undefined and size is the same as options[i], then print selected. same thing with 42-63 ^^ but as I said, you don't have to worry about it